third_party/terraform/utils/compute_instance_network_interface_helpers.go

```diff
@@ -0,0 +1,84 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	computeBeta "google.golang.org/api/compute/v0.beta"
+)
+
+func computeInstanceDeleteAccessConfigs(d *schema.ResourceData, config *Config, instNetworkInterface *computeBeta.NetworkInterface, project, zone, userAgent, instanceName string) error {
+	// TODO: This code deletes then recreates accessConfigs.  This is bad because it may
```

Comment:
This should probably be moved up to the site that calls delete before add, rather than the body of the delete fn